Zend/zend_operators.c

```diff
@@ -1984,6 +1984,9 @@ ZEND_API zend_result ZEND_FASTCALL concat_function(zval *result, zval *op1, zval
 
 		memcpy(ZSTR_VAL(result_str) + op1_len, Z_STRVAL_P(op2), op2_len);
 		ZSTR_VAL(result_str)[result_len] = '\0';
+		if (ZSTR_IS_VALID_UTF8(Z_STR_P(op1)) && ZSTR_IS_VALID_UTF8(Z_STR_P(op2))) {
```

Comment:
is this method called also from functions like `implode`?

also it would be great to test the flags by a test, if the flag is not available in userland, it should be possible to expose it in `zend_test`

Reply:
No, ``implode()`` uses a custom algorithm, similar to ``str_pad()`` adding those flags to these functions shouldn't be too difficult. Having a function in ``zend_test`` to check the flag is set is a good idea.

Reply:
I agree having a function in `zend_test` to test the flag is a good idea.